### PR TITLE
[AppService] Hotfix: Fix # 18266 - webapp config appsettings set command causing all values to default to "false" in version 2.24.0

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
#18265 , #18266

**Description**

When executing the webapp config appsettings set command the result Application Settings in the WebApp Configuration value is set to false for each setting defined in the settings array.
**Testing Guide**

JSON Settings Example
[
{
"slotSetting": false,
"name": "ExampleParameter1",
"value": "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter1)"
},
{
"slotSetting": false,
"name": " ExampleParameter2",
"value": "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter2)"
}
]

Command Example
az webapp config appsettings set --name $WebAppName --resource-group $ResourceGroup --settings @$settings.json
**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -270,7 +270,6 @@ def update_app_settings(cmd, resource_group_name, name, settings=None, slot=None
                 dest[setting_name] = value
                 result.update(dest)
 
-    result.update(slot_result)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:273]: Hmm how did the previous version regress this, was this added recently?

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 2]
```diff
@@ -617,17 +617,17 @@ def test_update_webapp_settings_thru_json(self, resource_group):
 
         self.assertEqual(output[0], {
             'name': 's',
-            'value': 'False',
+            'value': 'value',
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:620]: Do we have a test case with this JSON format for both app & slot. If not can you add one, so we don't miss this again? Thanks for taking care of this quickly.

**Review by panchagnula — CHANGES_REQUESTED** [review]
Can we add a test for slot as well if we don't already have? Otherwise LGTM!

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:273]: We have rewritten the logic to differentiate `settings` and `slotsettings` logic. This line should suppose to remove but mistakenly it did not happened.

**Comment by Kotasudhakarreddy** [comment]
> Can we add a test for slot as well if we don't already have? Otherwise LGTM!

Added test for Slots.

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py (hunk 2 above)
Comment (Kotasudhakarreddy) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:625]: Added

**Comment by Kotasudhakarreddy** [comment]
@Juliehzl please consider this for the hotfix release happens on 31/05(Shanghai time)

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:273]: Ok this was broken for a while now. Thanks for explaining.

**Review by panchagnula — CHANGES_REQUESTED** [review]
LGTM

**Review by Juliehzl — APPROVED** [review]

**Comment by rafaelberrocalj** [comment]
Hello there is another problem, if a comma is forgotten inside the array, all properties are created in the same key=value.

File example:

[
{
"slotSetting": false,
"name": "ExampleParameter1",
"value": "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter1)"
},
{
"slotSetting": false,
"name": " ExampleParameter2",
"value": "@Microsoft.KeyVault(SecretUri=https://xxxxx.vault.azure.net/secrets/ExampleParameter2)"
}, //COMMA 
]

Notice the comma in the last key/value setting, the result is something like this:
image

**Comment by Kotasudhakarreddy** [comment]
@rafaelberrocalj I am unable to repro this issue as CLI will not allow the invalid Json format.  please try the same in latest CLI version `2.25.0` and let us know if you still face the issue.

**Pull request merged**